streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -602,7 +607,7 @@ class StreamingContext private[streaming] (
     if (stopSparkContext) sc.stop()
     uiTab.foreach(_.detach())
     // The state should always be Stopped after calling `stop()`, even if we haven't started yet:
-    state = Stopped
+    state = STOPPED
```

Comment:
This is not a new change - but is the stop method ok to call over and over? For example, what happens if one of the methods above throws and the state never gets set? Is it OK to call stop on the other methods again and again? Maybe we should move the stop call to the first line in the `case STARTED` block